cpp/src/rolling/rolling_detail.cuh

```diff
@@ -166,31 +178,62 @@ struct DeviceRollingArgMinMax {
                              size_type end_index,
                              size_type current_index)
   {
-    using AggOp = typename corresponding_operator<op>::type;
-    AggOp agg_op;
+    if constexpr (std::is_same_v<InputType, cudf::string_view>) {
```

Comment:
The `operator()` function implementation for `struct_view` and `string_view` appear to be completely different as shown by this `if constexpr` expression. Also, the `string_view` path does not require the member variable `agg_op`. Since it's `operator()` and member variables do not align, perhaps they should not be defined in the same functor? Or maybe some other refactoring is possible?

Reply:
Update: I have separated the previous functor into 2 separate ones for string and struct.